pkg/apis/core/validation/validation.go

```diff
@@ -1973,6 +2081,63 @@ func ValidateVolumeMounts(mounts []core.VolumeMount, volumes sets.String, contai
 	return allErrs
 }
 
+func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]string, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	devicepath := sets.NewString()
+	devicename := sets.NewString()
+
+	if devices != nil && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("volumeDevices"), "Container volumeDevices is disabled by feature-gate"))
+		return allErrs
+	}
+	if devices != nil {
+		for i, dev := range devices {
+			idxPath := fldPath.Index(i)
+			devName := dev.Name
+			devPath := dev.DevicePath
+			didMatch, isPVC := isMatchedDevice(devName, volumes)
+			if len(devName) == 0 {
+				allErrs = append(allErrs, field.Required(idxPath.Child("name"), ""))
+			}
+			if devicename.Has(devName) {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), devName, "must be unique"))
+			}
+			// Must be PersistentVolumeClaim volume source
+			if didMatch && !isPVC {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), devName, "can only use volume source type of PersistentVolumeClaim for block mode"))
+			}
+			if !didMatch {
+				allErrs = append(allErrs, field.NotFound(idxPath.Child("name"), devName))
+			}
+			if len(devPath) == 0 {
+				allErrs = append(allErrs, field.Required(idxPath.Child("devicePath"), ""))
+			}
+			if devicepath.Has(devPath) {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("devicePath"), devPath, "must be unique"))
+			}
+			if !path.IsAbs(devPath) {
```

Comment:
this won't work for cross-platform nodes/containers. see discussion at https://github.com/kubernetes/kubernetes/pull/51240#issuecomment-342387620

Reply:
ack...fixed by stealing code from ```volumeMounts``` implementation of this validation check

Reply:
follow the resolution on that thread - https://github.com/kubernetes/kubernetes/pull/51240#issuecomment-342530520 ... I think the consensus is to drop the absolute validation server-side and force the path to be absolute in the kubelet

Reply:
cc @thockin

Reply:
ok, so if I understand correctly, I should remove the IsAbs check entirely and let the kubelet handle